### ISSUE TITLE
Cover Manager: current_cover_file could be removed

### ISSUE THREAD

**Issue by nightridefm** [savonet/liquidsoap#4005]
https://github.com/savonet/liquidsoap/blob/33a62b4858c0d1a9b442fe7aae13f7a0b43c89d0/src/libs/extra/metadata.liq#L28-L42

If there is a name collision with the incoming `filename` vs `last_cover_file()`, the current cover file will be removed and cause source errors

**Comment by vitoyucepi** [comment]
Hi @nightridefm,
Could you share the logs for this case?
As far as I remember the `filename` can be either `default` or the result `file.temp`.

**Comment by nightridefm** [comment]
@vitoyucepi it can, but if for whatever reason the previous file.temp name is the same as the next filename, it will be removed
at least `last_cover != default` should become `last_cover != default and last_cover != filename`

I am actually writing my own version that uses `file.exists()` as well, just in case.

Sample of logs spamming my server, burning CPU and causing catchup errors
```
2024/07/05 04:23:52 [request:3] Nonexistent file or ill-formed URI "/tmp/cover-manager85f0fe/cover_ada73c.jpeg"!
2024/07/05 04:23:52 [request:3] Nonexistent file or ill-formed URI "/tmp/cover-manager85f0fe/cover_ada73c.jpeg"!
2024/07/05 04:23:52 [request:3] Nonexistent file or ill-formed URI "/tmp/cover-manager85f0fe/cover_ada73c.jpeg"!
2024/07/05 04:23:52 [request:3] Nonexistent file or ill-formed URI "/tmp/cover-manager85f0fe/cover_ada73c.jpeg"!
2024/07/05 04:23:52 [request:3] Nonexistent file or ill-formed URI "/tmp/cover-manager85f0fe/cover_ada73c.jpeg"!
2024/07/05 04:23:52 [request:3] Nonexistent file or ill-formed URI "/tmp/cover-manager85f0fe/cover_ada73c.jpeg"!
2024/07/05 04:23:52 [request:3] Nonexistent file or ill-formed URI "/tmp/cover-manager85f0fe/cover_ada73c.jpeg"!
2024/07/05 04:23:52 [request:3] Nonexistent file or ill-formed URI "/tmp/cover-manager85f0fe/cover_ada73c.jpeg"!
2024/07/05 04:23:52 [request:3] Nonexistent file or ill-formed URI "/tmp/cover-manager85f0fe/cover_ada73c.jpeg"!
2024/07/05 04:23:52 [request:3] Nonexistent file or ill-formed URI "/tmp/cover-manager85f0fe/cover_ada73c.jpeg"!
2024/07/05 04:23:52 [request:3] Nonexistent file or ill-formed URI "/tmp/cover-manager85f0fe/cover_ada73c.jpeg"!
2024/07/05 04:23:52 [request:3] Nonexistent file or ill-formed URI "/tmp/cover-manager85f0fe/cover_ada73c.jpeg"!
2024/07/05 04:23:52 [request:3] Nonexistent file or ill-formed URI "/tmp/cover-manager85f0fe/cover_ada73c.jpeg"!
2024/07/05 04:23:52 [request:3] Nonexistent file or ill-formed URI "/tmp/cover-manager85f0fe/cover_ada73c.jpeg"!
```

**Comment by vitoyucepi** [comment]
I'm looking for logs with the label `cover` and levels 3 or 4.
Specifically, I need consecutive logs for https://github.com/savonet/liquidsoap/blob/33a62b4858c0d1a9b442fe7aae13f7a0b43c89d0/src/libs/extra/metadata.liq#L92-L97

**Comment by nightridefm** [comment]
I've got mine at level 3, but I think you actually want this at log level info.
https://github.com/savonet/liquidsoap/blob/33a62b4858c0d1a9b442fe7aae13f7a0b43c89d0/src/libs/extra/metadata.liq#L36-L39

I would have to revert my fixes and wait ~24hrs for this issue to occur again

**Comment by toots** [comment]
Should be fixed now, thanks for reporting!

**Comment by toots** [comment]
We decided to reopen and investigate more. Not sure that the original changes did address the underlying issue.

**Comment by toots** [comment]
@nightridefm have you actually experienced this issue? From the code, `last_cover_file` is always a fresh temporary file so this should never happen.

**Comment by vitoyucepi** [comment]
@toots, the issue was automatically closed because #4039 was merged.

**Comment by toots** [comment]
> @toots, the issue was automatically closed because #4039 was merged.

Yeah, I believe it should be fixed now. The cover is passed directly as a `request` and properly deleted when the request is done playing.
